### PR TITLE
added analytics documentation

### PR DESCRIPTION
**Pull request by rxiayth**
## Overview

Just adding analytics documentation to repo

### CODE REVIEW

**Comment by ruxandrafed** [comment]
Thanks SO MUCH for documenting this, @rxiayth! Here are a few suggestions for you:
- Could we add a bit more detail in the "How" section of the main analytics doc? Just a paragraph about the purpose of the data layer (without repeating what's in Confluence) and why we need to update things in it in the context of a SPA (single page app).
- I remember you told @nathanmarshall and I to wrap this code in a try/catch block just to ensure our experiences don't crash if the Adobe library isn't loaded for any reason. Could you please add that to the document, too?
- I would also recommend enhancing the example with the actual data layer changes (changing the content of the cart **before** calling the event. It helps re-enforce the importance of timing (first you modify the data layer object, then you tell Adobe it's time to read the updated data layer). 
- Also, who should tell us how we name the events? Where do `scAdd` and `add_to_cart` come from?
- Can you also add a note that this way to solve the problem does not apply for those using Adobe Launch already, and it's restricted to DTM?
- As we are still considering DTM as the de facto tool for this (because we're not all ready for Adobe Launch), I wonder if this is a good opportunity to mention the analytics scripts package we have (https://github.com/telus/analytics-scripts). I can do that after if you don't have time :)

Hope I didn't scare you with all of these suggestions :) Let me know what you think, please.

File: analytics/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+To measure KPIs we leverage Adobe Analytics for web and mobile app data collection across Telus. 
+
+
+## What
+Adobe Analytics provides the space to store, query and visualize data.
+Adobe DTM or Adobe Launch is a separate space, which specifies what and when data is sent to Adobe Analytics.
+(Adobe Launch is the tag manager we are migrating towards)
+
+## How
+In addition to the Adobe tools above, every page needs a 'dataLayer' object, which represents what data is available for sending to Adobe Analytics. As a tag manager like Adobe DTM or Adobe Launch is loaded on page, code will run to check for dataLayer and send a pageview as result. Nowadays, especially with SPA behavior, dataLayer needs to be updated as user interacts with page and cause DOM pages.
```
Comment (kspaans) [inline analytics/README.md]: Please use the monospace marketdown (backticks: `) when referring to code variables names like `dataLayer`.

File: analytics/adobe_analytics_functions.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Adobe Analytics Functions
+
+## Why
+Existing analytics tagging depends on an out-of-box Adobe functionality to detect when dataLayer object changes.
```
Comment (kspaans) [inline analytics/adobe_analytics_functions.md]: Formatting of `dataLayer`.

File: analytics/adobe_analytics_functions.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+This detection is a black-box and does not meet our requirements as users can trigger many changes quickly. 
+To avoid delaying when we set datalayer, we are exposing few methods to call to trigger hits directly.
+
+
+## Caveat
+This section will be updated as we extend more functionalities.
```
Comment (kspaans) [inline analytics/adobe_analytics_functions.md]: Functionality not functionalities.

File: analytics/adobe_analytics_functions.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+- Check page is loading **Adobe DTM**, and not Adobe Launch(look for scripts src containing `adobedtm` and not `launch`)
+  (we will enable the same function in Adobe Launch in near future)
+- Check dataLayer.product is populated as expected
```
Comment (kspaans) [inline analytics/adobe_analytics_functions.md]: Formatting of `dataLayer.product`.

File: analytics/adobe_analytics_functions.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+```javascript
+var s = _satellite.getToolsByType('sc')[0].getS();
+s.events="scRemove"
+s.track('remove_from_cart');
+```
+
+
+## Notes
+- Naming conventions for values inside s.events are all out-of-box and can't be changed. 
+- Naming conventions for values in s.track are customized and typically start with a verb.
```
Comment (kspaans) [inline analytics/adobe_analytics_functions.md]: Formatting of variable names on this line and the line above.

File: analytics/adobe_analytics_functions.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+To avoid delaying when we set datalayer, we are exposing few methods to call to trigger hits directly.
+
+
+## Caveat
+This section will be updated as we extend more functionalities.
+
+- Check page is loading **Adobe DTM**, and not Adobe Launch(look for scripts src containing `adobedtm` and not `launch`)
+  (we will enable the same function in Adobe Launch in near future)
+- Check dataLayer.product is populated as expected
+- Use try-catch just in case Adobe DTM is not loaded to prevent bad user experience(someone did not read ^^)
```
Comment (kspaans) [inline analytics/adobe_analytics_functions.md]: Please remove the parenthetical.

File: analytics/adobe_analytics_functions.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+- Ensure dataLayer is **updated first** before calling the functions below. 
+E.g., 
+	1. user lands on product detail page 
+	2. dataLayer is present with page level info
+	3. a pageview hit is sent  
+	4. user clicks to add product to cart
+	5. dataLayer is updated with product info
```
Comment (kspaans) [inline analytics/adobe_analytics_functions.md]: Formatting.

File: analytics/adobe_analytics_functions.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+This section will be updated as we extend more functionalities.
+
+- Check page is loading **Adobe DTM**, and not Adobe Launch(look for scripts src containing `adobedtm` and not `launch`)
+  (we will enable the same function in Adobe Launch in near future)
+- Check dataLayer.product is populated as expected
+- Use try-catch just in case Adobe DTM is not loaded to prevent bad user experience(someone did not read ^^)
+- Ensure dataLayer is **updated first** before calling the functions below. 
+E.g., 
+	1. user lands on product detail page 
+	2. dataLayer is present with page level info
```
Comment (kspaans) [inline analytics/adobe_analytics_functions.md]: Formatting.

File: analytics/adobe_analytics_functions.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+
+## Caveat
+This section will be updated as we extend more functionalities.
+
+- Check page is loading **Adobe DTM**, and not Adobe Launch(look for scripts src containing `adobedtm` and not `launch`)
+  (we will enable the same function in Adobe Launch in near future)
+- Check dataLayer.product is populated as expected
+- Use try-catch just in case Adobe DTM is not loaded to prevent bad user experience(someone did not read ^^)
+- Ensure dataLayer is **updated first** before calling the functions below.
```
Comment (kspaans) [inline analytics/adobe_analytics_functions.md]: Formatting.

**Comment by ruxandrafed** [comment]
Thanks for adding more details, @rxiayth! Please review Kyle's suggestions too when you have a chance. Thanks :)

File: analytics/adobe_analytics_functions.md (hunk 5 above)
Comment (rxiayth) [reply analytics/adobe_analytics_functions.md]: @kspaans Updated formatting. Thanks!

**Comment by rxiayth** [comment]
@kspaans Updated with changes, thanks!

**Comment by kspaans** [comment]
Is it necessary to add the HTML files? Also, there is a conflict with the top-level README now.

File: analytics/adobe_analytics_functions.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+- Check page is loading **Adobe DTM**, and not Adobe Launch(look for scripts src containing `adobedtm` and not `launch`)
+  (we will enable the same function in Adobe Launch in near future)
+- Check `dataLayer.product` is populated as expected
+- Use try-catch just in case Adobe DTM is not loaded to prevent bad user experience(someone did not read ^^)
+- Ensure there is no updates to `dataLayer.event` when using any functions below.
+- Check `dataLayer` is **updated first** before calling the functions below. 
+
+E.g., 
+	1. user lands on product detail page
```
[the remark this answers is not in the document]
Comment (rxiayth) [reply analytics/adobe_analytics_functions.md]: Please check again. Thanks

**Comment by kspaans** [comment]
Thanks @rxiayth!

**Pull request merged**